apps/chat/agent/tools.py

```diff
@@ -77,15 +74,25 @@ def action(
         message: str,
         **kwargs,
     ):
-        channel_session = ChannelSession.objects.filter(experiment_session=self.experiment_session).first()
-        PeriodicTask.objects.create(
-            name=f"reminder-{self.experiment_session.id}-{uuid.uuid4()}",
-            task=BOT_MESSAGE_FOR_USER_TASK,
-            kwargs=json.dumps({"chat_ids": [channel_session.external_chat_id], "message": message}),
-            clocked=ClockedSchedule.objects.create(clocked_time=datetime_due),
-            one_off=True,
-        )
+        periodic_task_kwargs = {
+            "clocked": ClockedSchedule.objects.create(clocked_time=datetime_due),
+            "one_off": True,
+        }
+        create_periodic_task(self.experiment_session, message=message, kwargs=periodic_task_kwargs)
         return "Success"
 
 
+def create_periodic_task(experiment_session: ExperimentSession, message: str, kwargs: dict):
```

Comment:
I'm not suggesting a change, just making note that another option here is to accept **kwargs and pass those through instead of accepting a dict. It can make it easier for the caller since they don't have to construct a dict.

Reply:
Hmm..good point. I'll make the update, it's small enough

Reply:
https://github.com/dimagi/open-chat-studio/pull/35/commits/738e80e08f9bb5e307cf3cdf329483c249f5af64